pinot-query-planner/pom.xml

```diff
@@ -49,6 +49,11 @@
       <groupId>org.codehaus.janino</groupId>
       <artifactId>commons-compiler</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.immutables</groupId>
+      <artifactId>value</artifactId>
```

Comment:
Do we need this? `value-annotations` should have all annotations

Reply:
Yes and no.

I think we don't even need `value-annotations` given the annotations are not kept at runtime.

The `value` dependency includes the annotation processor. There are 2 ways to add annotation processor in Maven:
1. The old way, just adding the annotation processor to the classpath (normally as provided, like I did here).
2. Include them in the `<annotationProcessor>` section of the compiler plugin.

The later is the recommended way, but we are still using the first one. They are incompatible, so we need to register all annotator processors using the newer way in the same PR. I can do that once this is merged.

Reply:
I've opened a PR using annotation processors in https://github.com/apache/pinot/pull/14185. If we merge that PR before this one we can modify this pom to use that mode as well, which will simplify this declaration.

Reply:
The PR has been merged. I've changed this code to use the same annotation as before